Sources/ManifestSerializer/JSONSerialization.swift

```diff
@@ -0,0 +1,130 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import PackageDescription
+
+/// A JSON representation of an element.
+public protocol JSONSerializable {
+    
+    /// Return a JSON representation.
+    func toJSON() -> AnyObject
+}
+
+public func jsonString(package: PackageDescription.Package) throws -> String {
+    
+    #if os(Linux)
+        let options: NSJSONWritingOptions = .PrettyPrinted
+    #else
+        let options: NSJSONWritingOptions = .prettyPrinted
+    #endif
+    
+    let json: AnyObject = package.toJSON()
+    let data = try NSJSONSerialization.data(withJSONObject: json, options: options)
+    guard let string = String(data: data, encoding: NSUTF8StringEncoding) else { fatalError("NSJSONSerialization emitted invalid data") }
+    return string
+}
+
+extension NSMutableDictionary {
+    static func withNew(block: @noescape (dict: NSMutableDictionary) -> ()) -> NSMutableDictionary {
+        let dict = NSMutableDictionary()
+        block(dict: dict)
+        return dict
+    }
+    
+    func set(object: AnyObject, forKey key: String) {
+        self[key.asNS()] = object
+    }
+}
+
+extension String {
+    public func asNS() -> NSString {
+        #if os(Linux)
+            return self.bridge()
+        #else
+            return self as NSString
+        #endif
+    }
+}
+
+extension Array {
+    public func asNS() -> NSArray {
+        #if os(Linux)
+            return self.bridge()
+        #else
+            return self.map { $0 as! AnyObject } as NSArray
```

Comment:
But then an array of incompatible structs would yield an empty NSArray, which is a silent failure. I think it's better to crash here so that it's clear it cannot be converted.